### PR TITLE
"WP API" -> "WordPress REST API" in README files

### LINKED ISSUE #2517: Change all instances of WP-API in readme
Our readme/description on the plugin repo has "WP-API" all through it. Let's fix all these up to match our real name, "WordPress REST API".

### PR DESCRIPTION
**Pull request by kadamwhite**
Fixes #2517, based on work by @davetgreen

This only changes content in prose (no change to links or changelog), and varies use of "WordPress REST API", "REST API" and "API" based on context to avoid distracting repetition.

### CODE REVIEW

**Comment by kadamwhite** [comment]
Looks like my editor changed the line endings on readme.txt or something; is that file auto-generated? I can run a script instead of making the edit in situ

File: README.md [hunk 1]
```diff
@@ -32,13 +32,9 @@ Update user with ID 4? Send a `POST` request to `/wp-json/wp/v2/users/4`. Get al
 posts with the search term "awesome"? `GET /wp-json/wp/v2/posts?filter[s]=awesome`.
 It's that easy.
 
-WP API exposes a simple yet easy interface to WP Query, the posts API, post
-meta API, users API, revisions API and many more. Chances are, if you can do
-it with WordPress, WP API will let you do it.
+The WordPress REST API exposes a simple yet easy interface to WP Query, the posts API, post meta API, users API, revisions API and many more. Chances are, if you can do it with WordPress, API will let you do it.
```
Comment (kadamwhite) [inline README.md]: Also grammar nit I only noticed in this view, maybe "simple yet powerful"? Simple yet easy seems redundant :)

**Comment by rmccue** [comment]
> Looks like my editor changed the line endings on readme.txt or something; is that file auto-generated? I can run a script instead of making the edit in situ

Hmm, not as far as I know. Can you revert it back?

File: README.md (hunk 1 above)
Comment (aaroncampbell) [reply README.md]: "simple yet powerful" sounds good. Alternatively we could just say "simple" or "easy to use"...but I feel like both of those are questionably accurate by themselves.

**Review by rmccue — APPROVED** [review]

**Pull request merged**